Task1

olegov-lab opened the pull request:
1. Task: https://github.com/rolling-scopes-school/tasks/blob/master/tasks/angular/intro.md
2. Screenshot:
![image](https://user-images.githubusercontent.com/70856734/161625633-5bc66d0e-a2f4-4e2f-8aae-977df20836e8.png)
3. Deploy: https://olegov-lab-youtube-client.netlify.app/

4. Done 21.03.2022 / deadline 25.03.2022

5. Score: 50 / 50
The app is generated (+5)
Added base project structure for components and models. ESLint configured (+10)
All the necessary components were created (+15)
All the necessary models were created. Search result list and search item components contain declarations of necessary fields (+20)

olegov-lab committed:
init:start task youtube-client -m 1) add eslint; 2) create components

olegov-lab committed:
feat:eslint correct

zheromskyV commented on youtube-client/tsconfig.json:
```diff
@@ -0,0 +1,32 @@
+    "enableI18nLegacyMessageIdFormat": false,
+    "strictInjectionParameters": true,
+    "strictInputAccessModifiers": true,
+    "strictTemplates": true
```
i recommend to add `"strictPropertyInitialization": false`

https://stackoverflow.com/questions/49699067/property-has-no-initializer-and-is-not-definitely-assigned-in-the-construc

olegov-lab replied:
fixed

zheromskyV commented on youtube-client/src/app/app-routing.module.ts:
```diff
@@ -0,0 +1,12 @@
+})
+export class AppRoutingModule { }
+
+export default AppRoutingModule;
```
## remove ALL `export default`

 you must not use it in Angular (unless you're **absolutely** sure you know what you're doing - e.g. you write some custom bundling logic for webpack, or smth like this)

you must always use only named exports (like on line 10 - `export class AppRoutingModule`)

default exports are not allowed with AOT (https://angular.io/guide/aot-compiler)
here it works because you have both named and default exports, so default one is ignored

olegov-lab replied:
fixed

zheromskyV commented on youtube-client/src/app/header/header.component.ts:
```diff
@@ -0,0 +1,16 @@
+  styleUrls: ['./header.component.scss'],
+})
+export class HeaderComponent implements OnInit {
+  constructor() { }
```
usually we remove empty methods, so in this case you'd have just `export class HeaderComponent {}`
but it's fine for now, no changes required

olegov-lab replied:
ok

zheromskyV commented on youtube-client/src/app/header/header.component.ts:
```diff
@@ -0,0 +1,16 @@
+  constructor() { }
+
+  /* eslint class-methods-use-this: ["error", { "exceptMethods": ["ngOnInit"] }] */
+  ngOnInit(): void {
```
not really a great practice to disable rules like this, but fine for now, no changes required
and again, normally there will be no empty methods at all

olegov-lab replied:
ok

zheromskyV commented on youtube-client/src/app/search/search-item/search-item.component.spec.ts:
```diff
@@ -0,0 +1,25 @@
+    await TestBed.configureTestingModule({
+      declarations: [SearchItemComponent],
+    })
+      .compileComponents();
```
i'd recommend to add `prettier` to your project, so that it would reformat your code in a consistent way on file save, it will be really helpful in the future, especially when working with rxjs, ngrx

but it's up to you

olegov-lab replied:
fixed

olegov-lab committed:
feat:Mentor's remarks corrected

-export default removed; add 'strictPropertyInitialization: false';-add prettier

olegov-lab committed:
feat:fixed mistake 'export default'
